### ISSUE TITLE
Support concatenated strings

### ISSUE THREAD

**Issue by sindresorhus** [dustinspecker/eslint-plugin-no-use-extend-native#1]
These currently aren't detected.

```js
('str' + 'ing').custom();
```

```js
(foo + 'string').custom();
```

*Doesn't matter what `foo` is as it will be coerced to a string.*
